docs/proposals/upstream-settings.md

```diff
@@ -0,0 +1,232 @@
+# Enhancement Proposal-2467: Upstream Settings Policy
+
+- Issue: https://github.com/nginxinc/nginx-gateway-fabric/issues/2467
+- Status: Implementable
+
+## Summary
+
+This Enhancement Proposal introduces the `UpstreamSettingsPolicy` API that allows Application developers to configure the behavior of the connection between NGINX and their upstream applications. This Policy will attach to a Service that is referenced in an HTTPRoute or GRPCRoute.
+
+## Goals
+
+- Define upstream settings.
+- Define an API for upstream settings.
+
+## Non-Goals
+
+- Provide implementation details for implementing the upstream settings policy.
+- Define an API for upstream settings for TLSRoute or other layer 4 routes.
+
+## Introduction
+
+### Upstream Settings
+
+Upstream settings are NGINX directives that affect requests sent from NGINX Gateway Fabric to an upstream application.
+
+To begin, the Upstream Settings Policy will include the following NGINX directives:
+
+- [`zone`](https://nginx.org/en/docs/http/ngx_http_upstream_module.html#zone)
+- [`keepalive`](https://nginx.org/en/docs/http/ngx_http_upstream_module.html#keepalive)
+- [`keepalive_requests`](https://nginx.org/en/docs/http/ngx_http_upstream_module.html#keepalive_requests)
+- [`keepalive_time`](https://nginx.org/en/docs/http/ngx_http_upstream_module.html#keepalive_time)
+- [`keepalive_timeout`](https://nginx.org/en/docs/http/ngx_http_upstream_module.html#keepalive_timeout)
+
+In the future, we can extend the Upstream Settings Policy to include more [upstream-related directives](nginx-extensions.md#upstream-settings).
+
+## API, Customer Driven Interfaces, and User Experience
+
+The `UpstreamSettingsPolicy` API is a CRD that is a part of the `gateway.nginx.org` Group. It adheres to the guidelines and requirements of a Direct Policy as outlined in the [Direct Policy Attachment GEP](https://gateway-api.sigs.k8s.io/geps/gep-2648/). It will target and be attached to a Service which is referenced in an HTTPRoute or GRPCRoute.
+
+Below is the Golang API for the `UpstreamSettingsPolicy` API:
+
+### Go
+
+```go
+package v1alpha1
+
+import (
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+type UpstreamSettingsPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of the UpstreamSettingsPolicy.
+    Spec UpstreamSettingsPolicySpec `json:"spec"`
+
+    // Status defines the state of the UpstreamSettingsPolicy.
+    Status gatewayv1alpha2.PolicyStatus `json:"status,omitempty"`
+}
+
+type UpstreamSettingsSpec struct {
```

Comment:
Is this supposed to be `UpstreamSettingsPolicySpec`?

Reply:
```suggestion
type UpstreamSettingsPolicySpec struct {
```